library/core/class.dispatcher.php

```diff
@@ -472,9 +472,9 @@ private function getEnabledApplicationFolders() {
      * @return array Returns an array in the form `[$controllerName, $parts]` where `$parts` is the remaining path parts.
      * If a controller cannot be found then an array in the form of `['', $parts]` is returned.
      */
-    private function findController($parts) {
+    private function findController(array $parts) {
```

Comment:
This is fine in this case (because this method is private), but I wanna point out that Vanilla is littered with (bad) array/object interchangeability that can make this dangerous in many situations. It's also not part of the goal of this PR and it's omitted from the description, which increases the risk.